addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -602,3 +345,194 @@ def action_see_accrual_plans(self):
+                        lt_info[1]['leaves_requested'] += data['virtual_leaves_taken']
+                        lt_info[1]['leaves_approved'] += data['leaves_taken']
+                        leave_allocations |= allocation
+                sorted_leave_allocations = leave_allocations.filtered('date_to').sorted(key='date_to')\
+                    + leave_allocations.filtered(lambda a: not a.date_to)\
+                    if leave_allocations else False
+                closest_allocation = sorted_leave_allocations[0] if sorted_leave_allocations else self.env['hr.leave.allocation']
+                closest_allocation_remaining = allocations_leaves_consumed[employee][leave_type][closest_allocation]['virtual_remaining_leaves']\
+                    if closest_allocation else 0
+                if closest_allocation.date_to:
+                    closest_allocation_expire = format_date(self.env, closest_allocation.date_to, date_format="MM/dd/yyyy")
+                    closest_allocation_duration =\
+                        employee.resource_calendar_id._attendance_intervals_batch(
+                            datetime.combine(closest_allocation.date_to, time.min).replace(tzinfo=pytz.UTC),
+                            datetime.combine(date, time.max).replace(tzinfo=pytz.UTC))\
+                        if leave_type.request_unit in ['hour']\
+                        else (closest_allocation.date_to - date).days + 1
+                else:
+                    closest_allocation_expire = False
+                    closest_allocation_duration = False
+                lt_info[1].update({
+                    'closest_allocation_remaining': closest_allocation_remaining,
+                    'closest_allocation_expire': closest_allocation_expire,
+                    'closest_allocation_duration': closest_allocation_duration,
+                })
+                allocation_data[employee].append(lt_info)
+        return allocation_data
+
+    def get_leaves_consumed(self, employees):
+        employees = employees if employees else self.env['hr.employee'].browse(self._get_contextual_employee_id())
+        leaves = self.env['hr.leave'].search([
+            ('holiday_status_id', 'in', self.ids),
+            ('employee_id', 'in', employees.ids),
+            ('state', 'in', ['confirm', 'validate1', 'validate']),
+        ]).sorted('date_from')
+        allocations = self.env['hr.leave.allocation'].search([
+            ('employee_id', 'in', employees.ids),
+            ('holiday_status_id', 'in', self.ids),
+            ('state', '=', 'validate'),
+        ])
+
+        # allocation_leaves_consumed is a dictionary to map the number of days/hours of leaves taken per allocation
+        # The structure is the following:
+        # - KEYS:
+        # allocation_leaves_consumed
+        #  |--employee_id
+        #      |--holiday_status_id
+        #          |--allocation
+        #              |--virtual_leaves_taken
+        #              |--leaves_taken
```

Comment:
Truly without really digging through the code, I don't get what is difference between virtual_leaves_taken and leaves_taken. We could explain it well :innocent: If I am not mistaken the difference is in state or so..one is for validated leaves, and one is for all.